### PR TITLE
#167190446 Implement follow feature

### PR DESCRIPTION
**Pull request by chuxmykel**
#### What does this PR do?
- This PR implements a feature to let users follow and unfollow each other. It also adds the user's followers and followings to the response gotten from the get profile endpoint.

#### Description of task to be completed.
- create UserFollower pivot table to associate user and follower
- add follow and unfollow feature
- fetch user followers and followings with profile
- display logged in user following status on viewed profiles

#### How should this Be manually tested?
- visit the staging link that will be generated automatically and hit the following endpoints
>- `GET /profiles/<username>` To get another user's profile
>- `GET /user` To get your own profile
>- `POST /profiles/:username/follow` To follow a user
>- `DELETE /profiles/:username/follow` To unfollow a user

#### What are the relevant Pivotal Tracker stories?
- [#167190446](https://www.pivotaltracker.com/story/show/167190446)

### CODE REVIEW

**Review by topseySuave — APPROVED** [review]

**Review by chiboycalix — APPROVED** [review]
LGTM. nice work @chuxmykel

**Review by cvjude — APPROVED** [review]

**Review by obayomi96 — APPROVED** [review]
Thanks for adding the app.json Variables for heroku. LGTM

File: server/controllers/userController.js [hunk 1]
```diff
@@ -279,28 +297,48 @@ class UserController {
     * @memberof UserController
     */
   static async getAuserProfile(req, res) {
-    const { username } = req.params;
-
-    const user = await models.User.findOne({ where: { username } });
-
-    if (!user) return errorStat(res, 404, 'No user found');
+    const { params: { username }, user } = req;
```
Comment (encodedBicoding) [inline server/controllers/userController.js:300]: Nice Destructuring

**Review by encodedBicoding — APPROVED** [review]
LGTM

File: server/controllers/userController.js (hunk 1 above)
Comment (chuxmykel) [reply server/controllers/userController.js:300]: Thanks bro.

**Review by Hadeneekeh — APPROVED** [review]

**Pull request merged**